tsdb/index/index.go

```diff
@@ -993,39 +972,126 @@ func (r *Reader) PostingsRanges() (map[labels.Label]Range, error) {
 	return m, nil
 }
 
-// ReadSymbols reads the symbol table fully into memory and allocates proper strings for them.
-// Strings backed by the mmap'd memory would cause memory faults if applications keep using them
-// after the reader is closed.
-func ReadSymbols(bs ByteSlice, version int, off int) ([]string, map[uint32]string, error) {
+type Symbols struct {
+	bs      ByteSlice
+	version int
+	off     int
+
+	index []int
```

Comment:
I feel it's quite cryptic. Maybe `chunkedOffsets`?

Reply:
Changed to offsets.